rpcs3/Input/product_info.cpp

```diff
@@ -0,0 +1,206 @@
+			CELL_PAD_PCLASS_PROFILE_DRUM_CYM_RIDE,
+		.capabilites = CELL_PAD_CAPABILITY_PS3_CONFORMITY | CELL_PAD_CAPABILITY_PRESS_MODE | CELL_PAD_CAPABILITY_HP_ANALOG_STICK | CELL_PAD_CAPABILITY_ACTUATOR | CELL_PAD_CAPABILITY_SENSOR_MODE
+	},
+	{
+		.type = product_type::dj_hero_turntable,
+		.class_id = CELL_PAD_PCLASS_TYPE_DJ,
+		.vendor_id = vendor_id::sony_cea,
+		.product_id = product_id::dj_hero_turntable,
+		.pclass_profile =
+			CELL_PAD_PCLASS_PROFILE_DJ_MIXER_ATTACK |
+			CELL_PAD_PCLASS_PROFILE_DJ_MIXER_CROSSFADER |
+			CELL_PAD_PCLASS_PROFILE_DJ_MIXER_DSP_DIAL |
+			CELL_PAD_PCLASS_PROFILE_DJ_DECK1_STREAM1 |
+			CELL_PAD_PCLASS_PROFILE_DJ_DECK1_STREAM2 |
+			CELL_PAD_PCLASS_PROFILE_DJ_DECK1_STREAM3 |
+			CELL_PAD_PCLASS_PROFILE_DJ_DECK1_PLATTER |
+			CELL_PAD_PCLASS_PROFILE_DJ_DECK2_STREAM1 |
+			CELL_PAD_PCLASS_PROFILE_DJ_DECK2_STREAM2 |
+			CELL_PAD_PCLASS_PROFILE_DJ_DECK2_STREAM3 |
+			CELL_PAD_PCLASS_PROFILE_DJ_DECK2_PLATTER,
+		.capabilites = CELL_PAD_CAPABILITY_PS3_CONFORMITY | CELL_PAD_CAPABILITY_PRESS_MODE | CELL_PAD_CAPABILITY_HP_ANALOG_STICK | CELL_PAD_CAPABILITY_ACTUATOR | CELL_PAD_CAPABILITY_SENSOR_MODE
+	},
+	{
+		.type = product_type::dance_dance_revolution_mat,
+		.class_id = CELL_PAD_PCLASS_TYPE_DANCEMAT,
+		.vendor_id = vendor_id::konami_de,
+		.product_id = product_id::dance_dance_revolution_mat,
+		.pclass_profile =
+			CELL_PAD_PCLASS_PROFILE_DANCEMAT_CIRCLE |
+			CELL_PAD_PCLASS_PROFILE_DANCEMAT_CROSS |
+			CELL_PAD_PCLASS_PROFILE_DANCEMAT_TRIANGLE |
+			CELL_PAD_PCLASS_PROFILE_DANCEMAT_SQUARE |
+			CELL_PAD_PCLASS_PROFILE_DANCEMAT_RIGHT |
+			CELL_PAD_PCLASS_PROFILE_DANCEMAT_LEFT |
+			CELL_PAD_PCLASS_PROFILE_DANCEMAT_UP |
+			CELL_PAD_PCLASS_PROFILE_DANCEMAT_DOWN,
+		.capabilites = CELL_PAD_CAPABILITY_PS3_CONFORMITY | CELL_PAD_CAPABILITY_PRESS_MODE | CELL_PAD_CAPABILITY_HP_ANALOG_STICK | CELL_PAD_CAPABILITY_ACTUATOR | CELL_PAD_CAPABILITY_SENSOR_MODE
+	},
+	{
+		.type = product_type::ps_move_navigation,
+		.class_id = CELL_PAD_PCLASS_TYPE_NAVIGATION,
+		.vendor_id = vendor_id::sony_corp,
+		.product_id = product_id::ps_move_navigation,
+		.pclass_profile = 0x0,
+		.capabilites = CELL_PAD_CAPABILITY_PS3_CONFORMITY | CELL_PAD_CAPABILITY_PRESS_MODE | CELL_PAD_CAPABILITY_HP_ANALOG_STICK | CELL_PAD_CAPABILITY_ACTUATOR | CELL_PAD_CAPABILITY_SENSOR_MODE
+	},
+	{
+		.type = product_type::ride_skateboard,
+		.class_id = CELL_PAD_PCLASS_TYPE_SKATEBOARD,
+		.vendor_id = vendor_id::sony_cea,
+		.product_id = product_id::ride_skateboard,
+		.pclass_profile = 0x0,
+		.capabilites = CELL_PAD_CAPABILITY_PS3_CONFORMITY | CELL_PAD_CAPABILITY_PRESS_MODE | CELL_PAD_CAPABILITY_SENSOR_MODE
+	},
+	{
+		.type = product_type::guncon_3,
+		.class_id = CELL_PAD_FAKE_TYPE_GUNCON3,
+		.vendor_id = vendor_id::namco,
+		.product_id = product_id::guncon_3,
+		.pclass_profile = 0x0,
+		.capabilites = 0x0
+	}
+	};
+
+	product_info get_product_info(product_type type)
+	{
+		for (const auto& p : input_products)
```

Comment:
why not just make it a map by type? there can only be one anyway.

Reply:
I changed it, but now the type is present twice, both used as a key and as a member.